api/proto/teleport/integration/v1/integration_service.proto

```diff
@@ -0,0 +1,88 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package teleport.integration.v1;
+
+import "google/protobuf/empty.proto";
+import "teleport/legacy/types/types.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/integration/v1";
+
+// IntegrationService provides methods to manage Integrations with 3rd party APIs.
+service IntegrationService {
+  // ListIntegrations returns a paginated list of Integration resources.
+  rpc ListIntegrations(ListIntegrationsRequest) returns (ListIntegrationsResponse);
+
+  // GetIntegration returns the specified Integration resource.
+  rpc GetIntegration(GetIntegrationRequest) returns (types.IntegrationV1);
+
+  // CreateIntegration creates a new Integration resource.
+  rpc CreateIntegration(CreateIntegrationRequest) returns (types.IntegrationV1);
+
+  // UpdateIntegration updates an existing Integration resource.
+  rpc UpdateIntegration(UpdateIntegrationRequest) returns (types.IntegrationV1);
+
+  // DeleteIntegration removes the specified Integration resource.
+  rpc DeleteIntegration(DeleteIntegrationRequest) returns (google.protobuf.Empty);
+
+  // DeleteAllIntegrations removes all Integrations.
+  rpc DeleteAllIntegrations(DeleteAllIntegrationsRequest) returns (google.protobuf.Empty);
+}
+
+// ListIntegrationsRequest is a request for a paginated list of Integrations.
+message ListIntegrationsRequest {
+  // Limit is the maximum amount of resources to retrieve.
+  int32 limit = 1;
+  // NextKey is the key for the next page of Integrations.
+  string next_key = 2;
+}
+
+// ListIntegrationsResponse a paginated list of Integrations.
+message ListIntegrationsResponse {
+  // Integrations is a list of Integrations.
+  repeated types.IntegrationV1 integrations = 1;
+  // NextKey is the key for the next page of Integrations.
+  string next_key = 2;
+  // TotalCount is the total number of resources available after filter, if any.
```

Comment:
i saw someone got confused by this explanation (they thought it meant the total in the page that we just retrieved `len(repeated types.IntegrationV1)`), i'm not sure if the below is any better.

```suggestion
  // TotalCount is the total number of integrations in all pages after filter, if any.
```